src/layout/verticalPanel/allApplicationList.ts

```diff
@@ -0,0 +1,181 @@
+/** Gnome libs imports */
+import * as Clutter from 'clutter';
+import { Text } from 'clutter';
+import * as GObject from 'gobject';
+import * as Shell from 'shell';
+import { registerGObjectClass } from 'src/utils/gjs';
+import * as St from 'st';
+import { SearchResultEntry } from './searchResultList';
+const ParentalControlsManager = imports.misc.parentalControlsManager;
+
+/** Extension imports */
+const Me = imports.misc.extensionUtils.getCurrentExtension();
+
+@registerGObjectClass
+export class AllApplicationList extends St.BoxLayout {
+    static metaInfo: GObject.MetaInfo = {
+        GTypeName: 'AllApplicationList',
+        Signals: {
+            'result-selected-changed': {
+                param_types: [SearchResultEntry.$gtype],
+                accumulator: 0,
+            },
+        },
+    };
+    searchEntry: St.Entry;
+    text: Text;
+    parentalControlsManager;
+    entrySelected: SearchResultEntry | null = null;
+    constructor(searchEntry: St.Entry) {
+        super({
+            style_class: 'search-result-list',
+            vertical: true,
+        });
+        this.searchEntry = searchEntry;
+        this.text = this.searchEntry.clutter_text;
+        // Note: Clutter typedefs seem to be incorrect. According to the docs `ev` should be a Clutter.KeyEvent, but it actually seems to be a Clutter.Event.
+        this.text.connect('key-press-event', this.onKeyPress.bind(this));
+        this.parentalControlsManager = ParentalControlsManager.getDefault();
+        this.parentalControlsManager.connect(
```

Comment:
Does this have to be disconnected when the list is destroyed? I'm not sure if `getDefault` refers to a global instance or not.

Reply:
probably needed to be destroyed either way